modules/extensions/src/api/experimental/exec.ts

```diff
@@ -52,28 +37,59 @@ export async function exec(
     })
   );
 
-  const result: Promise<SeparatedOutputExecResult | CombinedOutputExecResult> =
-    new Promise(async (resolve) => {
-      const { exitCode, error } = await promise;
+  let dispose = async () => {
+    (await execResult).dispose();
+  };
+
+  const resultPromise = new Promise<SpawnResult>(async (resolve) => {
+    const { exitCode, error } = await (await execResult).promise;
 
-      if (options.separateStdErr) {
-        resolve({
-          stdOut: outputStr,
-          stdErr: errorStr,
-          error,
-          exitCode: exitCode ?? 0,
-        });
-      } else {
-        resolve({
-          error,
-          output: outputStr,
-          exitCode: exitCode ?? 0,
-        });
-      }
+    resolve({
+      error,
+      exitCode: exitCode ?? 0,
     });
+  });
 
   return {
-    result,
+    resultPromise,
     dispose,
   };
 }
+
+/**
+ * Executes a command in the shell, with given arguments and environment variables
+ */
+async function exec(
+  command: string,
+  options: {
+    env?: Record<string, string>;
+  } = {}
+): Promise<ExecResult> {
+  let stderr = "";
+  let stdout = "";
+  const { resultPromise } = spawn({
+    args: ["bash", "-c", command],
+    env: options.env ?? {},
+    splitStderr: true,
```

Comment:
Wdyt about not doing this? I've rarely had to use this in reality. Also when in buffering mode, the splitting loses a lot of information (like the sequence of the outputs)

Reply:
so we just return output and exitCode?